File: src/android/app/src/main/java/org/citra/citra_emu/utils/InsetsHelper.java
package org.citra.citra_emu.utils;

import android.content.Context;
import android.content.res.Resources;
import android.view.ViewGroup;

import androidx.core.graphics.Insets;

import com.google.android.material.appbar.AppBarLayout;

public class InsetsHelper {
    public static final int THREE_BUTTON_NAVIGATION = 0;
    public static final int TWO_BUTTON_NAVIGATION = 1;
    public static final int GESTURE_NAVIGATION = 2;

    public static void insetAppBar(Insets insets, AppBarLayout appBarLayout)
    {
        ViewGroup.MarginLayoutParams mlpAppBar =
                (ViewGroup.MarginLayoutParams) appBarLayout.getLayoutParams();
        mlpAppBar.leftMargin = insets.left;
        mlpAppBar.rightMargin = insets.right;
        appBarLayout.setLayoutParams(mlpAppBar);
    }

    public static int getSystemGestureType(Context context) {
        Resources resources = context.getResources();
        int resourceId = resources.getIdentifier("config_navBarInteractionMode", "integer", "android");
        if (resourceId != 0) {
            return resources.getInteger(resourceId);
        }
        return 0;
    }
}

File: src/android/app/src/main/java/org/citra/citra_emu/utils/InsetsHelper.kt
// Copyright 2023 Citra Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.

package org.citra.citra_emu.utils

import android.annotation.SuppressLint
import android.content.Context

object InsetsHelper {
    const val THREE_BUTTON_NAVIGATION = 0
    const val TWO_BUTTON_NAVIGATION = 1
    const val GESTURE_NAVIGATION = 2

    @SuppressLint("DiscouragedApi")
    fun getSystemGestureType(context: Context): Int {
        val resources = context.resources
        val resourceId = resources.getIdentifier(
            "config_navBarInteractionMode",
            "integer",
            "android"
        )
        return if (resourceId != 0) resources.getInteger(resourceId) else 0
    }
}
